tutorials/dapt-curation/code/downloaders.py

```diff
@@ -0,0 +1,254 @@
+    extractor = GitHubExtractor()
+
+    output_format = {
+        "text": str,
+        "id": str,
+        "file_extension": str,
+        "file_type": str,
+        "category": str,
+        "line_count": int,
+        "size_in_bytes": int,
+        "path": str,
+    }
+
+    dataset = download_and_extract(
+        urls=urls,
+        output_paths=[
+            os.path.join(output_jsonl_dir, os.path.basename(url)) for url in urls
+        ],
+        downloader=downloader,
+        iterator=iterator,
+        extractor=extractor,
+        output_format=output_format,
+        keep_raw_download=True,
+    )
+    # Force the computation of the dataset
+    dataset.persist()
+    return output_jsonl_dir
+
+
+def download_pdf_sources(
+    source_links_file: Optional[str] = None,
+    output_dir: Optional[str] = None,
+    limit: Optional[int] = None,
+):
+    """
+    Downloads Arxiv Pdf sources specified in a file and extracts them.
+
+    Args:
+        source_links_file: Path to the file containing the Arxiv PDF source links.
+        output_dir: Directory where the downloaded and extracted files will be stored.
+        limit: Maximum number of pdf sources to download and extract.
+
+    Raises:
+        FileNotFoundError: If the source_links_file does not exist.
+
+    Returns:
+        str: Path to the output directory where the downloaded and extracted files are stored.
+    """
+
+    if source_links_file is None:
+        source_links_file = os.path.join(
+            os.path.dirname(__file__), "sources", "arxiv_urls.jsonl"
+        )
+
+    if not os.path.exists(source_links_file):
+        raise FileNotFoundError(f"File '{source_links_file}' not found.")
+
+    urls = pd.read_json(path_or_buf=source_links_file, lines=True)
+    urls = urls[0].tolist()
+
+    if limit:
+        urls = urls[:limit]
+
+    if output_dir is None:
+        output_dir = os.path.join(
+            os.path.dirname(__file__), "data", "raw", "arxiv_pdfs"
+        )
+
+    print(urls)
+    os.makedirs(output_dir, exist_ok=True)
+    output_jsonl_dir = os.path.join(output_dir, "jsonl")
+    os.makedirs(output_jsonl_dir, exist_ok=True)
+
+    downloader = ArxivDownloader(output_dir)
+    iterator = ArxivIterator()
+    extractor = ArxivExtractor()
+
+    output_format = {
+        "text": str,
+        "id": str,
+        "file_extension": str,
+        "file_type": str,
+        "category": str,
+        "line_count": int,
+        "size_in_bytes": int,
+        "path": str,
+    }
+
+    dataset = download_and_extract(
+        urls=urls,
+        output_paths=[
+            os.path.join(output_jsonl_dir, os.path.basename(url)) for url in urls
+        ],
+        downloader=downloader,
+        iterator=iterator,
+        extractor=extractor,
+        output_format=output_format,
+        keep_raw_download=True,
+    )
+    # Force the computation of the dataset
+    dataset.persist()
+    return output_jsonl_dir
+
+
+if __name__ == "__main__":
```

Comment:
We might want to remove this as well, since we were just using it for our own testing.

Reply:
done